### PR TITLE
First stab at docs.

### PR DESCRIPTION
**Pull request by vext01**
Here is a README.md how to build rust and solutions to some common problems.

I've also mentioned why this script is needed at all.

Have I missed anything?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+build a beta rust, then (if you want a nightly) it uses this beta rust to build
+a nightly rust. It also deals with ensuring the right cargo version is used at
+each stage.
+
+## Quick Start
+
+First:
+
+```
+$ ./build init
```
Comment (ltratt) [inline README.md]: `./build.sh`?

Comment (vext01) [inline README.md]: yep

**Comment by vext01** [comment]
@semarie If you are OK with this, let me squash before the merge please. Thanks.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+## Why is this Script Needed?
+
+### Reason 1: Cross compiling.
+
+Under normal circumstances, rust upstream would cross compile beta and nightly
+rust on their Linux machines, making the resulting binaries available via
+`rustup`. However, OpenBSD has a modified linker meaning that cross compiled
+binaries do not run.
+
+There are ongoing efforts to use clang/LLVM as OpenBSD's base compiler and
```
Comment (semarie) [inline README.md]: the `use clang/LLVM as OpenBSD's base compiler` isn't accurate at the moment.

clang is mostly use for `arm64` port, but as several developers use it to crosscompile from OpenBSD to OpenBSD-arm64, the compiler gains better supports for crosscompiling OpenBSD.

File: README.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+```
+
+## Why is this Script Needed?
+
+### Reason 1: Cross compiling.
+
+Under normal circumstances, rust upstream would cross compile beta and nightly
+rust on their Linux machines, making the resulting binaries available via
+`rustup`. However, OpenBSD has a modified linker meaning that cross compiled
+binaries do not run.
```
Comment (semarie) [inline README.md]: I am unsure about "do not run". it seems a bit categorical. Saying it is complex is more neutral.

File: README.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+If you encounter this error, you probably want to use `eg++` as the linker
+instead of the base `g++`. A convenient way to do this is to have your project
+use `eg++` if the host system is OpenBSD. For example, for an amd64 system, add
+in `.cargo/config`:
+
+```
+[target.x86_64-unknown-openbsd]
+linker = "eg++"
+```
+
```
Comment (semarie) [inline README.md]: Adding a section on configuration and default values ?

File: README.md [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+upstream master branch, or you can use the bootstrap cargo to install a new one
+in `~/.cargo/bin`:
+
+```
+$ cargo install cargo
+```
+
+If you want that as your default cargo, don't forget to add it to the `PATH`.
+
+## I get an Error About Linking -lstdc++ Failing.
```
Comment (vext01) [inline README.md]: No, it's only occurs when building user programs using the environment set up by build-rust. I will remove it for now.

**Comment by vext01** [comment]
Try this

File: README.md [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+Where `<target>` is either `beta` or `nightly`. If you choose `nightly` but
+have not yet built  `beta`, then `beta` will be built first automatically.
+
+Once this is done you will have a working Rust environment (including cargo) in
+`install_dir/<target>`. If you want this to be your default rustc and cargo,
+then you probably want to add `install_dir/<target>/bin` to your `PATH` in your
+shell rc.
+
+Rust encodes an rpath by default, so you should *not* have to set an
+`LD_LIBRARY_PATH`.
```
Comment (semarie) [inline README.md]: if I agree on the theory (`LD_LIBRARY_PATH` shouldn't be need due to rpath), it is a point that is wrong currently.

in fact, the build script replace `bin/rustc` by a shell wrapper to workaround that.

I need to check the problem to see if it is in `ld.so` (partial supports of `$ORIGIN` for example) or in rust side.

Comment (vext01) [inline README.md]: Ah. I see. I guess I could just remove this line?

Comment (semarie) [inline README.md]: yes please

File: README.md [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+$ ./build.sh init
+```
+
+This step uses the `SUDO` command you chose in the first step to install some
+packages. Please read the script to ensure you are happy with the commands that
+will be run as root.
+
+Now run:
+```
+$ ./build <target>
```
Comment (semarie) [inline README.md]: ./build.sh

**Comment by semarie** [comment]
@vext01 I squashed and merged. Thanks !

**Pull request merged**